lib/vanity/adapters/abstract_adapter.rb

```diff
@@ -48,72 +48,81 @@ def destroy_metric(metric)
 
       # -- Experiments --
 
-      def experiment_persisted?(experiment)
+      def experiment_persisted?(experiment_id)
         fail "Not implemented"
       end
 
       # Store when experiment was created (do not write over existing value).
-      def set_experiment_created_at(experiment, time)
+      def set_experiment_created_at(experiment_id, time)
         fail "Not implemented"
       end
 
       # Return when experiment was created.
-      def get_experiment_created_at(experiment)
+      def get_experiment_created_at(experiment_id)
         fail "Not implemented"
       end
 
       # Returns true if experiment completed.
-      def is_experiment_completed?(experiment)
+      def is_experiment_completed?(experiment_id)
         fail "Not implemented"
       end
 
       # Store whether an experiment is enabled or not
-      def set_experiment_enabled(experiment, enabled)
+      def set_experiment_enabled(experiment_id, enabled)
         fail "Not implemented"
       end
 
       # Returns true if experiment is enabled, the default (Vanity.configuration.experiments_start_enabled) is true.
       # (*except for mock_adapter, where default is true for testing)
-      def is_experiment_enabled?(experiment)
+      def is_experiment_enabled?(experiment_id)
         fail "Not implemented"
       end
 
-      # Returns counts for given A/B experiment and alternative (by index).
+      # Returns counts for given A/B experiment (actual object) and alternative (by index).
       # Returns hash with values for the keys :participants, :converted and
       # :conversions.
       def ab_counts(experiment, alternative)
         fail "Not implemented"
       end
 
+      # Returns counts by metric for a given A/B experiment (actual object) & alternative (by index).
+      # Returns a hash with keys metric id and values that are hashes like {added_to_cart: {converted: x, conversions: y}}
+      # Keep in mind that "converted" is unique conversions while "conversions" is a total tally (or
+      # you could set it up to track dollar sales or something like that)
+      # only really supported for redis adapter
+      def ab_counts_by_metric(experiment, alternative)
```

Comment:
Can you add this new method to `doc/adapters.textile`? We wouldn't need to include which adapters support it there.